allensdk/brain_observatory/behavior/data_objects/timestamps/stimulus_timestamps/stimulus_timestamps.py

```diff
@@ -80,17 +91,121 @@ def from_sync_file(
             return cls.from_sync_file(
                         sync_file=sync_file,
-                        monitor_delay=dict_repr['monitor_delay'])
+                        monitor_delay=monitor_delay)
         else:
             stim_file = BehaviorStimulusFile.from_json(dict_repr=dict_repr)
             return cls.from_stimulus_file(
                         stimulus_file=stim_file,
-                        monitor_delay=dict_repr['monitor_delay'])
+                        monitor_delay=monitor_delay)
+
+    @classmethod
+    def from_multiple_stimulus_blocks(
+            cls,
+            sync_file: SyncFile,
+            list_of_stims: List[Union[BehaviorStimulusFile,
+                                      MappingStimulusFile,
+                                      ReplayStimulusFile]],
+            stims_of_interest: Optional[List[int]] = None,
+            monitor_delay: float = 0.0,
+            frame_time_lines: Union[str, List[str]] = 'vsync_stim',
+            frame_time_line_direction: str = 'rising',
+            frame_count_tolerance: float = 0.0) -> "StimulusTimestamps":
+        """
+        Construct a StimulusTimestamps instance by registering
+        multiple stimulus blocks to one sync file and concatenating the results
+
+        Parameters
+        ----------
+        sync_file: SyncFile
+
+        list_of_stims: List[Union[BehaviorStimulusFile,
+                                  MappingStimulusFile,
+                                  ReplayStimulusFile]]
+            The list of StimulusFiles to be registered to the SyncFile
+            **in the order that they were presented to the mouse**
+
+        stims_of_interest: Optional[List[int]]
+            The indexes in list_of_stims of the timestamps to be
+            concatenated into this one StimulusTimestamps object.
+            If `None` (default), the timestamps from all stimulus files
+            are concatenated
+
+        monitor_delay: float
+            in seconds
+
+        frame_time_lines: Union[str, List[str]]
+            The line to be used to find raw frame times (usually 'vsync_stim').
+            If a list, the code will scan the list in order until a line
+            that is present in the sync file is found. That line will be used.
+
+        frame_time_line_direction: str
+            Either 'rising' or 'falling' indicating which edge
+            to use in finding the raw frame times
+
+        frame_count_tolerance: float
+            The tolerance to within two blocks of frame counts are considered
+            equal
+        """
+        behavior_stimulus_files = [x for x in list_of_stims
+                                   if isinstance(x, BehaviorStimulusFile)]
+        if len(behavior_stimulus_files) == 0:
+            raise ValueError(
+                'One of the values in `list_of_stims` must be a '
+                '`BehaviorStimulusFile`')
+        elif len(behavior_stimulus_files) > 1:
```

Comment:
So we can only have one Behave stim but having multiples of other types makes sense? Is this correct?

Reply:
Technically having multiple of another doesn't make sense. I just added this check specifically for behavior stim because we pass that one to the class constructor. The others are just used to construct the timestamps.